python/pyspark/sql/udf.py

```diff
@@ -129,18 +127,12 @@ def _create_py_udf(
             else useArrow
         )
     regular_udf = _create_udf(f, returnType, PythonEvalType.SQL_BATCHED_UDF)
-    return_type = regular_udf.returnType
     try:
         is_func_with_args = len(getfullargspec(f).args) > 0
     except TypeError:
         is_func_with_args = False
-    is_output_atomic_type = (
-        not isinstance(return_type, StructType)
-        and not isinstance(return_type, MapType)
-        and not isinstance(return_type, ArrayType)
-    )
     if is_arrow_enabled:
-        if is_output_atomic_type and is_func_with_args:
+        if is_func_with_args:
```

Comment:
Just curious, why can't we enable arrow optimization for UDFs that do not have input arguments?

Reply:
Arrow optimization only makes sense when UDFs have inputs because it's an improvement on input (de)serialization.

Reply:
But does it hurt if we enable Arrow for functions without input arguments? Then we can eliminate this restriction and enable Arrow for all kinds of Python UDFs.

Reply:
Good point. The warning will be removed when we enable Arrow by default.